### PR TITLE
ABNF updates

### PR DESCRIPTION
**Pull request by ashnazg**
(redo PR #97 from @fleshgrinder about ABNF)

### CODE REVIEW

**Comment by ashnazg** [comment]
Ping @ondrejmirtes @muglug @neuro159 @mindplay-dk @GaryJones @mvriel @jaapio for opinions.

**Comment by jaapio** [comment]
I don't understand the recommendation about used characters. However I never used any other characters it is the responsibility of the developer to decide how to name their elements. As long it flows the requirements stated in the php manual. I think that should be the only restriction. The recommendation not needed in my opinion.

File: proposed/phpdoc.md [hunk 1]
```diff
@@ -166,34 +166,53 @@ interpreted as described in [RFC 2119][RFC2119].
...
+        (?<FUNCTION>(?&FQN)(?:::(?&NAME))?\(\))
+        (?<VARIABLE>(?&FQN)::\$(?&NAME))
+        (?<CONSTANT>(?&FQN)::(?&NAME))
+        (?<FQN>(?:\\(?&NAME))+)
+        (?<NAME>[[:alpha:]_\x7f-\xff][[:alnum:]_\x7f-\xff]*)
+      )
+      ^(?&FQSEN)$/x
+
+  The ABNF restricts all structural element names to the byte range that is
+  given in the [basics PHP manual page][PHP_BASICS]. However, it is highly
```
Comment (GaryJones) [inline proposed/phpdoc.md]: I'm not seeing a link to `PHP_BASICS`?

**Comment by ashnazg** [comment]
If someone feels strongly about keeping this change, please bring it up on the mailing list for discussion.

**Pull request closed without merging**